src/main/kotlin/no/nav/omsorgspenger/OppgaveClient.kt

```diff
@@ -0,0 +1,166 @@
+package no.nav.omsorgspenger
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.SerializationFeature
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.nimbusds.jwt.SignedJWT
+import io.ktor.client.HttpClient
+import io.ktor.client.features.ResponseException
+import io.ktor.client.request.accept
+import io.ktor.client.request.get
+import io.ktor.client.request.header
+import io.ktor.client.request.post
+import io.ktor.client.statement.HttpResponse
+import io.ktor.client.statement.HttpStatement
+import io.ktor.http.ContentType
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.contentType
+import io.ktor.util.toByteArray
+import no.nav.helse.dusseldorf.ktor.health.HealthCheck
+import no.nav.helse.dusseldorf.ktor.health.Healthy
+import no.nav.helse.dusseldorf.ktor.health.UnHealthy
+import no.nav.helse.dusseldorf.oauth2.client.AccessTokenClient
+import no.nav.helse.dusseldorf.oauth2.client.CachedAccessTokenClient
+import no.nav.k9.rapid.river.Environment
+import no.nav.k9.rapid.river.hentRequiredEnv
+import no.nav.omsorgspenger.oppgave.Oppgave
+import no.nav.omsorgspenger.oppgave.OppgaveRespons
+import no.nav.omsorgspenger.oppgave.oppdatertOppgaveBody
+import org.slf4j.LoggerFactory
+
+internal class OppgaveClient(
+        private val env: Environment,
+        accessTokenClient: AccessTokenClient,
+        private val httpClient: HttpClient
+) : HealthCheck {
+    private val cachedAccessTokenClient = CachedAccessTokenClient(accessTokenClient)
+    private val baseUrl = env.hentRequiredEnv("OPPGAVE_BASE_URL")
+    private val oppgaveScopes = setOf(env.hentRequiredEnv("OPPGAVE_SCOPES"))
+    private val pingUrl = "$baseUrl/isReady"
+
+    internal suspend fun hentOppgave(correlationId: String, aktoerId: String, journalpostIder: Set<String>): OppgaveLøsning {
+        val payload = """
+            {
+            "journalpostId": "$journalpostIder",
```

Comment:
Er dette riktig? jsonen vil vel bli sende noe slik ut?

```json
{
   "journalpostId": "["1"]"
}
```

Skal ikke journalpostId være en array, ikke en String med array i?

Reply:
Nei det var feil. Ersatt med en JSONArray som inte er satt in i en string.